### ISSUE TITLE
Blocking cookies breaks withDarkMode component

### ISSUE THREAD

**Issue by aswanson-nr** [newrelic/opensource-website#795]
## Description

The `withDarkMode` component uses the localStorage API directly, which causes errors when the browser blocks access to those APIs. This causes most of the site to not display correctly.
<img width="1099" alt="Screen Shot 2021-06-01 at 11 33 44 AM" src="https://user-images.githubusercontent.com/70179303/120374011-d86c6a00-c2cd-11eb-8b9d-6bb6996a2796.png">

`withDarkMode` provides information about the current mode of the theme to various components around the site. It does this through checking localStorage for the `darkMode` key and using the `use-dark-mode` hook. When cookies are blocked by the browser, it blocks our access to certain browser APIs such as localStorage and causes this component to throw an uncaught error. This error comes from our attempt to access localStorage https://github.com/newrelic/opensource-website/blob/develop/src/components/withDarkMode.js#L10 and the `use-dark-mode` hook.

#### Solution
1. Wrap the [call to localStorage](https://github.com/newrelic/opensource-website/blob/develop/src/components/withDarkMode.js#L10) in a try/catch.
2. Tell `use-dark-mode` to use an in-memory store when localStorage isn't available, similar to how the [theme does it]( https://github.com/newrelic/gatsby-theme-newrelic/blob/develop/packages/gatsby-theme-newrelic/src/components/DarkModeToggle.js#L12).
## Steps to Reproduce

Run the opensource site and block all cookies on your browser.

## Expected Behavior
The core content loads without needing access to the localStorage.

**Comment by jpvajda** [comment]
@aswanson-nr will this bug be resolved with #794?

**Comment by aswanson-nr** [comment]
@jpvajda This will not be resolved by #794. That PR fixes the overall issues introduced through the theme, unfortunately there are still some problems with the opensource-site that won't be resolved. I broke this out into another ticket because I didn't want to keep blindly hacking at it.
I will take the action to add AC!

**Comment by elaguerta** [comment]
I'm starting to dig into this one.

**Comment by rudouglas** [comment]
Sorry @elaguerta didn't see your comment, ignore my PR 😬

**Comment by elaguerta** [comment]
@rudouglas I'm taking myself off this one. I put a `closes` keyword on your PR, but if you get a chance you may want to relink it. No worries at all - you didn't step on my toes, and I was glad to see your solution. Thanks for coordinating!
